### PR TITLE
Change ByteString.size from fun to val

### LINKED ISSUE #428: ByteString.size
I’d like to change this from a fun to a val for Kotlin.

### PR DESCRIPTION
**Pull request by Egorand**
Closes #428

### CODE REVIEW

File: okio/jvm/src/main/java/okio/ByteString.kt [hunk 1]
```diff
@@ -227,7 +227,11 @@ internal constructor(
   operator fun get(index: Int): Byte = getByte(index)
 
   /** Returns the number of bytes in this ByteString. */
-  open fun size() = data.size
+  val size
+    @JvmName("size") get() = getSize()
+
+  // Hack to work around Kotlin's limitation for using JvmName on open/override vals/funs
+  protected open fun getSize() = data.size
```
Comment (Egorand) [inline okio/jvm/src/main/java/okio/ByteString.kt]: On a second thought, `internal` might be a better option? This way it shouldn't appear in KDoc

**Comment by Egorand** [comment]
Japicmp: `methodExcludes = ['okio.ByteString#size()', 'okio.SegmentedByteString#size()']`

**Comment by swankjesse** [comment]
Is the exclusion for final/non-final?

**Comment by Egorand** [comment]
@swankjesse in `SegmentedByteString` it's for "REMOVED METHOD: PUBLIC(-) int size()", which makes sense, since this method is not overridden in that class anymore - it shouldn't manifest in the bytecode if it's inherited from the parent, right?

**Comment by swankjesse** [comment]
Gotcha. SegmentedByteString is not a public type, so I think this is a mistake in japicmp.

**Pull request merged**